ui/js/controllers/logviewer.js

```diff
@@ -210,11 +210,29 @@ logViewerApp.controller('LogviewerCtrl', [
                 });
             });
 
+            // Make the log and job artifacts available
             ThJobArtifactModel.get_list({job_id: $scope.job_id, name__in: 'text_log_summary,Job Info'})
             .then(function(artifactList) {
                 artifactList.forEach(function(artifact) {
                     if (artifact.name === 'text_log_summary') {
                         $scope.artifact = artifact.blob;
+                        $scope.step_data = artifact.blob.step_data;
+                        var initialStep = $scope.artifact.step_data.steps[0];
+
+                        // If the log contains no errors load the head otherwise
+                        // load the first failure step line in the artifact
+                        if (initialStep) {
```

Comment:
And we make sure one exists so we don't attempt to navigate to a step when its undefined.